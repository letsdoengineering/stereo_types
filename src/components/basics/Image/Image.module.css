.size-S {
    height: auto;
    width: 19rem;
}

.size-M {
    height: auto;
    width: 38rem;
}

.size-L {
    height: auto;
    width: 75rem;
}
