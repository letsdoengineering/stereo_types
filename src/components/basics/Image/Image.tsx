import React from 'react'
import getClassNames from '../../../utils/get-class-names'
import * as imageStyles from './Image.module.css'

type ImageProps = {
    alt: string
    src: string
    size?: 'S' | 'M' | 'L'
    className?: string
}

const Image: React.FC<ImageProps> = ({ alt, src, size = 'M', className }: ImageProps) => {
    const classNames = getClassNames({
        defaultClasses: [imageStyles[`size${size}`]],
        className,
    })
    return <img alt={alt} src={src} className={classNames} />
}

export default Image
