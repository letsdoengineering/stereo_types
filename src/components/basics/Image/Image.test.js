import * as React from 'react'
import Image from './Image'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'

describe('Image', () => {
    it('has not changed the JSX layout of the page from previous snapshot', () => {
        const tree = renderer
            .create(<Image alt={'Some text'} src={'Some url'} size={'M'} />)
            .toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('renders the page DOM that is accessible', async () => {
        const { container } = render(<Image alt={'Some text'} src={'Some url'} size={'M'} />)
        const results = await axe(container)

        expect(results).toHaveNoViolations()
    })
})
