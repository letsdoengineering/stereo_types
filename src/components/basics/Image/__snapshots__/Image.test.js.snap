// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image has not changed the JSX layout of the page from previous snapshot 1`] = `
<img
  alt="Some text"
  className="sizeM"
  src="Some url"
/>
`;
